packages/vue/src/dialog/use-dialog.ts

```diff
@@ -1,19 +1,19 @@
 import { connect, machine, type Context } from '@zag-js/dialog'
 import { normalizeProps, useMachine } from '@zag-js/vue'
-import { computed } from 'vue'
-import { transformComposableProps, useId } from '../utils'
+import { computed, reactive } from 'vue'
+import { useId } from '../utils'
 
 export type UseDialogProps = {
   context: Omit<Context, 'id'>
   emit: CallableFunction
 }
 
-export const useDialog = (props: UseDialogProps) => {
-  const { context, emit } = transformComposableProps(props)
```

Comment:
@Shyrro `transformComposableProps` was supposed to return a reactive context and still be able to destructure.

Reply:
Hey @TylerAPfledderer !
The transform composable does this : 

```
export function transformComposableProps<T extends { context: object }>(props: T): T {
  return {
    ...props,
    context: reactive(props.context),
  }
}
```

The object returned is not `reactive`. The context is. This means you can destructure context, not the resulting object. And destructuring the result will result in the context losing its reactivity.

For this to work, all the resulting object should be wrapped in the `reactive` function. Also, inside that function, `props` shouldn't be destructured to spread them. if you want to spread them, you should do `...reactive(props)`.

Hope that answers your question !

Reply:
@Shyrro understood!

With the refactor you did for the dialog composable, I'm finding this function unnecessary. The point of it was to simplify a pattern that was occuring.

I think that it should be removed for all composables.

And overall I would say that `defaultValue` would not be a prop to apply for some of the composables if `modelValue` is the preferred choice. (Since the ref applied to `modelValue` can take an initial or default value)

So each composable would have at least `emit` and `context` props and applied in the way you have done in this PR.

What do you think?

Reply:
@TylerAPfledderer , since we talked about this in Slack i'm merging this ;) 
We'll do the refactor later